src/ServicePulse.Host/app/js/views/monitored-endpoints/monitored-endpoints.controller.js

```diff
@@ -13,12 +13,9 @@
         });
 
         function updateUI() {
-            monitoringService.getRaw().then(function (data) {
-                $scope.endpoints = data["NServiceBus.Endpoints"];
-
-                timeoutId = $timeout(function () {
-                    updateUI();
-                }, 5000);
+            monitoringService.getEndpoints().subscribe(function (endpoint) {
+                $scope.endpoints = $scope.endpoints || {};
+                $scope.endpoints[endpoint.endpointName] = endpoint;
```

Comment:
not too familiar with rxJs, but pretty sure in rx.net you could merge different streams into a single result. so instead of producing from:
[a1, a2, a3] , [b1, b2] -> [a1, a2, a3, b1, b2]
you would produce
[a1, a2, a3], [b1, b2] -> [[a1, a2, a3, b1, b2]]
this would still have the disadvantage that you trigger complete re-rendering on each http request, but clearly less than with the current approach.